From 8a27be0451c657639bd9c3f0d1b94cdb82db0d4f Mon Sep 17 00:00:00 2001
From: Richard W.M. Jones <rjones@redhat.com>
Date: Thu, 19 Dec 2013 08:21:53 +0000
Subject: [PATCH 18/67] run: Use timeout --foreground option.

If timeout doesn't have this option (RHEL 6) don't use timeout at all.

Attempt to fix RHBZ#1025269.
---
 run.in |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/run.in b/run.in
index d82ba56..c6bc9c4 100755
--- a/run.in
+++ b/run.in
@@ -186,13 +186,13 @@ timeout_kill=30s
 
 # Do we have Padraig's timeout utility (from coreutils)?
 if timeout --help >/dev/null 2>&1; then
-    # Does this version of timeout have the -k option?  (Not on RHEL 6)
-    if timeout -k 10s 10s true >/dev/null 2>&1; then
-        timeout_k_opt="-k $timeout_kill"
+    # Must use the --foreground option (RHBZ#1025269).
+    if timeout --foreground 2 sleep 0 >/dev/null 2>&1; then
+        # Does this version of timeout have the -k option?  (Not on RHEL 6)
+        if timeout -k 10s 10s true >/dev/null 2>&1; then
+            timeout="timeout -k $timeout_kill $timeout_period"
+        fi
     fi
-    # Timeout (SIGTERM) after $timeout_period.
-    # Then send a second SIGKILL $timeout_kill seconds later.
-    timeout="timeout $timeout_k_opt $timeout_period"
 fi
 
 pid=$$
-- 
1.7.1

